107/src/test/java/org/ehcache/ParsesConfigurationExtensionTest.java

```diff
@@ -80,7 +82,7 @@ public void testXmlExample() throws ClassNotFoundException, SAXException, Instan
       // Test the config
       {
         final CacheRuntimeConfiguration<Long, Product> runtimeConfiguration = productCache.getRuntimeConfiguration();
-        assertThat(runtimeConfiguration.getResourcePools().getPoolForResource(ResourceType.Core.HEAP).getSize(), equalTo(200L));
+        assertThat(((SizedResourcePool)runtimeConfiguration.getResourcePools().getPoolForResource(ResourceType.Core.HEAP)).getSize(), equalTo(200L));
```

Comment:
This feels weird... I wonder if we can add generics to the ResourceType so that necessary cast be internalized and tied to the type of the resource being requested?